mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte

```diff
@@ -26,6 +27,23 @@
   ];
 
   let activeTab: TabItem;
+
+  const tabularData = getTabularDataStoreFromContext();
+  $: ({ selection } = $tabularData);
+  $: ({ selectedCells } = selection);
+
+  $: {
+    // Explicit dependency
+    $selectedCells;
+
+    if (selection.isAnyColumnCompletelySelected()) {
+      [, activeTab] = tabs;
+    }
+
+    if (selection.isAnyRowCompletelySelected()) {
+      [, , activeTab] = tabs;
+    }
+  }
```

Comment:
I'd like to avoid this reactive block if possible, it's quite inefficient since it has to run for every change in selection. Since we control the selection logic, we could have a separate class to manage the Inspector and control it as needed, in a ad-hoc manner. We have something similar implemented for the Data Explorer's inspector.